modules/common/topology/types.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2025 Red Hat
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package topology
```

Comment:
@olliewalsh @stuggi I thought that `topology` was better than having `topologyref` as package name. Because we're using references as input to retrieve and return a `Topology` object I think `topology` as package name applies here, which is different from what we use from `infra-operator`, where we refer to `topologyv1` as an API to call functions that acts on the CRD.